lib/assert.js

```diff
@@ -30,45 +30,49 @@ function test(ok, opts) {
 	}
 }
 
+function stringMsg(msg) {
+	return msg instanceof Error ? msg.message : typeof msg === "string" ? msg : msg ? JSON.stringify(msg) : null;
```

Comment:
Following your comment, I can think of two possible ways to go about this.
Throw an exception when msg is anything other than

1.  A string or 
2. A string or Error object (since for errro type,  the message propoerty value can be used).